src/test/java/com/imagination/cbs/service/impl/EmailServiceImplTest.java

```diff
@@ -0,0 +1,335 @@
+		verify(config).getTemplate("email.approval_request.ftl");
+		verify(approvalStatusDmRepository).findById(201L);
+	}
+
+	@Test
+	public void shouldSendEmailForBookingApprovalWithEmptyStringsWhenObjectsInBookingRevisionAreNull() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException, MessagingException {
+
+		MailRequest emailRequest = createMailRequest();
+		BookingRevision bookingRevision = createBookingRevisionWithNullObjects();
+		ReflectionTestUtils.setField(emailServiceImpl, "baseUrl", "https://imaginagtion.com/bookingSummary?bookingId={bookingId}&status={status}");
+		
+		when(config.getTemplate("email.approval_request.ftl")).thenReturn(template);
+		when(approvalStatusDmRepository.findById(201L)).thenReturn(Optional.empty());
+		
+		emailServiceImpl.sendEmailForBookingApproval(emailRequest, bookingRevision, "approval_request");
+
+		verify(config).getTemplate("email.approval_request.ftl");
+		verify(approvalStatusDmRepository).findById(201L);
+	}
+
+	@Test
+	public void shouldLogExceptionMessageOccuredDuringEmailSending_sendEmailForBookingApproval() {
+
+		MailRequest emailRequest = createMailRequest();
+		BookingRevision bookingRevision = createBookingRevisionWithNullObjects();
+		ReflectionTestUtils.setField(emailServiceImpl, "baseUrl", "https://imaginagtion.com/bookingSummary?bookingId={bookingId}&status={status}");
+		
+		emailServiceImpl.sendEmailForBookingApproval(emailRequest, bookingRevision, "approval_request");
+	}
+	
+	@Test
+	public void shouldSendContractorReceiptEmail() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException {
+		
+		BookingRevision bookingRevision = createBookingRevisionForContractReceipt();
+		CBSUser cbsUser = new CBSUser("Akshay");
+		cbsUser.setEmail("akshay@imagination.com");
+		
+		when(config.getTemplate("email.contractreceipt.ftl")).thenReturn(template);
+		when(loggedInUserService.getLoggedInUserDetails()).thenReturn(cbsUser);
+		
+		emailServiceImpl.sendContractReceipt(bookingRevision);
+		
+		verify(config).getTemplate("email.contractreceipt.ftl");
+		verify(loggedInUserService).getLoggedInUserDetails();
+	}
+	
+	@Test
+	public void shouldLogExceptionMessageOccuredDuringEmailSending_sendContractReceipt() {
+		
+		BookingRevision bookingRevision = createBookingRevisionForContractReceipt();
+		
+		emailServiceImpl.sendContractReceipt(bookingRevision);
+	}
+	
+	@Test
+	public void shouldSendInternalResourceEmail() throws TemplateNotFoundException, MalformedTemplateNameException, ParseException, IOException {
+		
+		InternalResourceEmailDto internalResourceEmailDto = createInternalResourceEmailDto();
+		
+		when(config.getTemplate("email.internalsource.ftl")).thenReturn(template);
+		
+		emailServiceImpl.sendInternalResourceEmail(internalResourceEmailDto);
+
+		verify(config).getTemplate("email.internalsource.ftl");
+	}
+	
+	@Test
+	public void shouldLogExceptionMessageOccuredDuringEmailSending_sendInternalResourceEmail() {
```

Comment:
no assert and no verify, what exactly we are testing here.

Reply:
This are actually to cover catch block in a code

Reply:
ahh... so it just for executing catch block for code coverage but does not verify or test anything... what do you do after catching the exception? i guess logging it .... can you try to verify loggers log method with same message ? give it a try if possible.

Reply:
Okk, i will try it!

Reply:
I gave a try, but since logger is private static and final so It can not be mocked.

Reply:
ok np ... just make sure that it gets called so it can come under code coverage.